src/corecel/data/Collection.hh

```diff
@@ -127,6 +127,41 @@ using ItemId = OpaqueId<T, size_type>;
 template<class T, class Size = size_type>
 using ItemRange = Range<OpaqueId<T, Size>>;
 
+//---------------------------------------------------------------------------//
+/*! Access data in a Range<T2> with an index of type T1
```

Comment:
```suggestion
/*!
 * Access data in a Range<T2> with an index of type T1.
```

Also surely there's a way to enforce this template matching? Maybe either C++20 concepts or a helper template struct (in the detail namespace):
```c++
template<class T>
struct IsOpaqueId { constexpr bool value = false; };
template<class V, class S>
struct IsOpaqueId<OpaqueId<V, S>> { constexpr bool value = true; }
```

and `static_assert(IsOpaqueId<T1>::value, "T1 isn't opaque ID");` and same for T2.

Reply:
I think `T1` needs to be an `OpaqueId` but `T2` can actually be anything

Reply:
To semantically be an "item map" though, they both need to be opaque IDs.